crates/oxc_transformer/src/proposals/explicit_resource_management.rs

```diff
@@ -158,10 +158,31 @@ impl<'a> Traverse<'a> for ExplicitResourceManagement<'a, '_> {
     /// ```
     fn exit_static_block(&mut self, block: &mut StaticBlock<'a>, ctx: &mut TraverseCtx<'a>) {
         let scope_id = block.scope_id();
-        if let Some(replacement) =
-            self.transform_statements(&mut block.body, None, scope_id, scope_id, ctx)
+        if let Some((new_stmts, needs_await, using_ctx)) =
+            self.transform_statements(&mut block.body, scope_id, ctx)
         {
-            block.body = ctx.ast.vec1(replacement);
+            let static_block_new_scope_id = ctx.scoping.insert_scope_between(
+                ctx.scoping().scope_parent_id(scope_id).unwrap(),
+                scope_id,
+                ScopeFlags::ClassStaticBlock,
+            );
+
+            ctx.scoping_mut().set_symbol_scope_id(using_ctx.symbol_id, static_block_new_scope_id);
+            ctx.scoping_mut().move_binding(scope_id, static_block_new_scope_id, &using_ctx.name);
+            ctx.scoping_mut().scope_flags_mut(scope_id).remove(ScopeFlags::ClassStaticBlock);
```

Comment:
I could be wrong, but I think this is equivalent, and would be slightly faster (unconditional write, instead of read + calculate + write):

```suggestion
            *ctx.scoping_mut().scope_flags_mut(scope_id) = ScopeFlags::StrictMode;
```

Reply:
yep this is the same, will update